custom_components/ha-google-home/__init__.py

```diff
@@ -47,18 +47,16 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
 
     session = async_get_clientsession(hass, verify_ssl=False)
 
+    zeroconf_instance = await zeroconf.async_get_instance(hass)
     glocaltokens_client = GlocaltokensApiClient(
-        hass, username, password, session, android_id
+        hass, username, password, session, android_id, zeroconf_instance
     )
 
-    zeroconf_instance = await zeroconf.async_get_instance(hass)
     coordinator = DataUpdateCoordinator(
         hass,
         _LOGGER,
         name=SENSOR,
-        update_method=lambda: glocaltokens_client.get_google_devices_information(
-            zeroconf_instance
-        ),
+        update_method=lambda: glocaltokens_client.update_google_devices_information(),
```

Comment:
```suggestion
        update_method=glocaltokens_client.update_google_devices_information,
```

Reply:
Wasn't able to drop `lambda`, it was throwing the error

Reply:
What error? Make sure you're passing the function itself, not calling it.

Reply:
Oh, flip you are totally right, i was calling the function. Thanks for the spot